### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading '_hasPreviewFlag')

### ISSUE THREAD

**Issue by outbreak** [prisma/prisma#14548]
### Bug description

I try use prisma transaction and have error:

```
<project-home>/src/database/client/runtime/index.js:23284
  const hasPreview = engine._hasPreviewFlag("tracing");
                            ^

TypeError: Cannot read properties of undefined (reading '_hasPreviewFlag')
    at getTracingConfig (<project-home>/src/database/client/runtime/index.js:23284:29)
    at Object.$transaction [as transaction] (<project-home>/src/database/client/runtime/index.js:29563:29)
```

### How to reproduce

Simple code in my handler:

```js
await prisma.$transaction([
  prisma.passwordOfAdministrator.upsert({
    where: {
      administratorId: confirmation.administratorId,
    },
    create: {
      administratorId: confirmation.administratorId,
      hash: password.hash,
      salt: password.salt,
      createdAt,
      updatedAt,
    },
    update: {
      hash: password.hash,
      salt: password.salt,
      updatedAt,
    },
  }),
  prisma.recoveryPasswordConfirmation.update({
    where: {
      id: confirmation.id,
    },
    data: {
      ip,
      agent,
      confirmAt,
    },
  }),
])
```

### Expected behavior

Execute `prisma.$transaction` without error:

```
TypeError: Cannot read properties of undefined (reading '_hasPreviewFlag')
```

### Prisma information

```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
  output   = "./client"
  engineType = "binary"
}
```

or 

```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
  output   = "./client"
  engineType = "library"
}
```

### Environment & setup

- OS: [Mac OS]
- Database: [PostgreSQL]
- Node.js version: [v16.15.0]

### Prisma Version

```
prisma                  : 4.1.1
@prisma/client          : 4.1.1
Current platform        : darwin
Query Engine (Node-API) : libquery-engine 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash    : 8d8414deb360336e4698a65aa45a1fbaf1ce13d8
Studio                  : 0.469.0
```

**Comment by matthewmueller** [comment]
This should be fixed in 4.2.0 going out next tuesday!

**Comment by schongabor** [comment]
+1
